platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/MapView.java

```diff
@@ -425,6 +425,7 @@ public void onDestroy() {
 
     if (mapboxMap != null) {
       mapboxMap.onDestroy();
+      mapboxMap = null;
```

Comment:
is there a concrete reason why we are explicitly setting mapboxMap to null? We treat mapboxMap as non null in our code (after initialisaton) and we potentially can end up in a codepath and run into a nullpointer (while destroying the map)

Reply:
I'm just thinking that after the destroy, mapboxMap is not usable anyway so would like to set it to null and if users try to `getMapAsync`, it won't be passed back to user again.
After consideration, I will remove this line to avoid possible null pointer issue.

Reply:
if we were able to pass by reference this setup would be great, end users that have MapboxMap as a field in their class would see that becoming null and the nullpointerexception would originate from their own code. While now, with java only being able to pass by value, you are marking mapoxMap eligble for garbage collection in MapView but lifetime of MapboxMap will be tied to the Activity hosting the MapView/MapboxMap. If they hit an nullpointer now, to them this seems like a mapbox internal issue. In any case, current setup doesn't feel ideal, let's fix that with a rewrite!